lib/ring_logger/client.ex

```diff
@@ -500,4 +525,9 @@ defmodule RingLogger.Client do
       []
     end
   end
+
+  @spec has_metadata?(RingLogger.entry(), atom(), any()) :: boolean()
+  def has_metadata?(%{metadata: metadata}, key, match_value) do
+    Keyword.has_key?(metadata, key) and Keyword.get(metadata, key) == match_value
```

Comment:
Good point - my requirement is an exact match, but I'm happy to extend this function to include a regexp possibility for the value. It would make sense considering I called the function grep :D

Reply:
This still isn't quite right. If Regex is used, then it will fail here. I think this can be adjusted to match a few conditions
```suggestion
    case metadata[key] do
      nil -> false
      val when is_binary(val) -> val =~ match_val
      val when val == match_val -> true
      _ -> false
    end
```

Reply:
Oops good catch, I'll add tests.